src/light_data/light_store.rs

```diff
@@ -106,7 +120,11 @@ impl<F: LurkField> LightStore<F> {
 
         // TODO: this needs to bail on encountering an opaque pointer
         while let Some(LightExpr::SymCons(s, ss)) = self.get(&ptr).flatten() {
-            let string = self.insert_scalar_string(s, store)?;
+            let string = if s == ScalarPtr::from_parts(ExprTag::Str, F::zero()) {
+                Ok(String::new())
+            } else {
+                self.insert_scalar_string(s, store)
+            }?;
```

Comment:
I think it might be clearer to move this conditional into `insert_scalar_string`. Also, isn't the immediate data (empty string) supposed to be 'added to the ScalarStore as the `LightStore` is traversed.'?